sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,410 @@
+                else:
+                    key_serialized = None
+                value = self.generate_json_with_schema_value(self.current_key)
+                value_serialized = self.value_serializer(value, SerializationContext(self.topic, MessageField.VALUE))
+
+            self.producer.produce(self.topic, value=value_serialized, key=key_serialized, callback=self.delivery_report)
+            self.current_key += 1
+            self.producer.poll(0)
+
+            if self.timeout and (time.time() - start_time) >= self.timeout:
+                LOGGER.warning("Duration exceeded, stopping Kafka producer thread.")
+                self.stop()
+                break
+
+        self.producer.flush()
+
+    def generate_avro_key(self, seed: int) -> dict | int | float | str | bool | bytes | None:
+        if self.key_schema:
+            key_schema = json.loads(self.key_schema)
+            return self.generate_data_from_schema(key_schema, seed)
+        return None
+
+    def generate_avro_value(self, seed: int) -> dict | int | float | str | bool | bytes | None:
+        if self.value_schema:
+            value_schema = json.loads(self.value_schema)
+            return self.generate_data_from_schema(value_schema, seed)
+        return None
+
+    def generate_json_with_schema_value(self, seed: int) -> str:
+        schema = json.loads(self.value_schema)
+        payload = self.generate_data_from_schema(schema, seed)
+
+        value_record = {
+            "schema": schema,
+            "payload": payload
+        }
+        value_str = json.dumps(value_record)
+        return value_str
+
+    def generate_json_with_schema_key(self, seed: int) -> str:
+        schema = json.loads(self.key_schema)
+        payload = self.generate_data_from_schema(schema, seed)
+
+        key_record = {
+            "schema": schema,
+            "payload": payload
+        }
+        key_str = json.dumps(key_record)
+        return key_str
+
+    def stop(self) -> None:
+        self.termination_event.set()
+
+    def kill(self) -> None:
+        self.stop()
+
+    def verify_results(self) -> (list[dict | None], list[str | None]):
+        self.join(self.timeout)
+        errors = []
+        if self.failed_count > 0:
+            msg = f"{self.failed_count} records failed to deliver."
+            TestFrameworkEvent(source=self.__class__.__name__, message=msg, severity=Severity.ERROR).publish()
+            errors.append(msg)
+        return [{'produced_count': self.produced_count}], errors
+
+
+class KafkaValidatorThread(KafkaThreadBase):
+    """
+    Thread to validate that records added to Kafka are correctly written to ScyllaDB.
+    """
+
+    def __init__(self, tester, params: SCTConfiguration, timeout: int, **kwargs):
+        super().__init__(tester, params, timeout, **kwargs)
+        self.mismatches = []
+        self.batch_size = int(kwargs.get('batch-size', 1000))
```

Comment:
Please add documentation on params for these threads. E.g. what is batch_size.

Reply:
Added doc. strings.